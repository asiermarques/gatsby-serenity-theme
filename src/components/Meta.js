import React from 'react';
import Helmet from 'react-helmet';

export default ({ title, description }) =>
        <Helmet>
            <title>{title}</title>
            <meta name="description" content={description} />
            <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/>
        </Helmet>