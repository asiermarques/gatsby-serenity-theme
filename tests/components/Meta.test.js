import React from 'react';
import renderer from 'react-test-renderer';
import Meta from '../../src/components/Meta';
import {describe, it, jest} from '@jest/globals';
import useSiteMetadata from '../../src/hooks/use-site-metadata';
import siteMetadataStub from '../sitemeta.stub';

jest.mock('../../src/hooks/use-site-metadata');


describe("Meta component", () => {
    it('renders correctly with all params', () => {
        useSiteMetadata.mockReturnValue(siteMetadataStub);

        const tree = renderer
            .create(<Meta
                metaTitle={"The Title On The Meta"}
                metaDescription={"The Description on the meta"}
                coverImage={"/image-custom.png"}
                urlPath={"/test"}/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
        expect(useSiteMetadata).toBeCalled();
    });

    it('renders correctly with default site params', () => {
        useSiteMetadata.mockReturnValue(siteMetadataStub);

        const tree = renderer
            .create(<Meta
                metaTitle={"The Title On The Meta"}
                metaDescription={"The Description on the meta"}/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
        expect(useSiteMetadata).toBeCalled();
    });
})
