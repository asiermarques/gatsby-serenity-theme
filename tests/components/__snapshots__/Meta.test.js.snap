// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta component renders correctly with all params 1`] = `
Array [
  <title>
    The Title On The Meta
  </title>,
  <link
    href="http://test.com/test"
    rel="canonical"
  />,
  <meta
    content="The Description on the meta"
    name="description"
  />,
  <meta
    content="width=device-width, initial-scale=1, shrink-to-fit=no"
    name="viewport"
  />,
  <link
    href="/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    href="/site.webmanifest"
    rel="manifest"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="The site title"
    property="og:title"
  />,
  <meta
    content="The Description on the meta"
    property="og:description"
  />,
  <meta
    content="http://test.com/test"
    property="og:url"
  />,
  <meta
    content="/image-custom.png"
    property="og:image"
  />,
  <meta
    content="The site title"
    property="og:site_name"
  />,
  <meta
    content="https://asiermarques.com"
    property="article:author"
  />,
]
`;

exports[`Meta component renders correctly with default site params 1`] = `
Array [
  <title>
    The Title On The Meta
  </title>,
  <link
    href="http://test.com"
    rel="canonical"
  />,
  <meta
    content="The Description on the meta"
    name="description"
  />,
  <meta
    content="width=device-width, initial-scale=1, shrink-to-fit=no"
    name="viewport"
  />,
  <link
    href="/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    href="/site.webmanifest"
    rel="manifest"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="The site title"
    property="og:title"
  />,
  <meta
    content="The Description on the meta"
    property="og:description"
  />,
  <meta
    content="http://test.com"
    property="og:url"
  />,
  <meta
    content="/the-image.png"
    property="og:image"
  />,
  <meta
    content="The site title"
    property="og:site_name"
  />,
  <meta
    content="https://asiermarques.com"
    property="article:author"
  />,
]
`;
